sdk/core/core-http/src/serializer.ts

```diff
@@ -733,7 +778,12 @@ function serializeCompositeType(
   return object;
 }
 
-function getXmlObjectValue(propertyMapper: Mapper, serializedValue: any, isXml: boolean) {
+function getXmlObjectValue(
+  propertyMapper: Mapper,
+  serializedValue: any,
+  isXml: boolean,
+  options?: { xmlCharKey?: string }
```

Comment:
same thought, why make this optional everywhere? We should make it optional where it comes in from a public interface, but required everywhere else so we don't have default logic sprinkled everywhere.